src/uct/base/uct_mem.c

```diff
@@ -165,6 +168,7 @@ ucs_status_t uct_mem_alloc(size_t length, const uct_alloc_method_t *methods,
                 ucs_assert(memh != UCT_MEM_HANDLE_NULL);
                 mem->md       = md;
                 mem->mem_type = mem_type;
+                mem->sys_dev  = sys_dev;
```

Comment:
Please add a test or extend an existing test in `uct/test_mem` to check `uct_allocated_memory_t::sys_dev`.